haas/ext/obm/ipmi.py

```diff
@@ -87,6 +87,15 @@ def power_off(self):
             raise OBMError('Could not power off node %s', self.label)
 
     @no_dry_run
+    def set_bootdev(self, dev):
+        valid_bootdevices = ['disk', 'pxe', 'none']
+        if dev not in valid_bootdevices:
+            raise OBMError('Invald boot device')
```

Comment:
Create a new exception OBMValidationError, so you can distinguish between HIL failing to set the boot device (500), and the boot device not being valid (400).